examples/layers/widgets/gestures.dart

```diff
@@ -104,42 +125,15 @@ class _GestureDemoState extends State<GestureDemo> {
 
   void _handleColorChange() {
     setState(() {
-      if (_swatch == Colors.blueGrey)
-        _swatch = Colors.red;
-      else if (_swatch == Colors.red)
-        _swatch = Colors.pink;
-      else if (_swatch == Colors.pink)
-        _swatch = Colors.purple;
-      else if (_swatch == Colors.purple)
-        _swatch = Colors.deepPurple;
-      else if (_swatch == Colors.deepPurple)
-        _swatch = Colors.indigo;
-      else if (_swatch == Colors.indigo)
-        _swatch = Colors.blue;
-      else if (_swatch == Colors.blue)
-        _swatch = Colors.lightBlue;
-      else if (_swatch == Colors.lightBlue)
-        _swatch = Colors.cyan;
-      else if (_swatch == Colors.teal)
-        _swatch = Colors.green;
-      else if (_swatch == Colors.green)
-        _swatch = Colors.lightGreen;
-      else if (_swatch == Colors.lightGreen)
-        _swatch = Colors.lime;
-      else if (_swatch == Colors.lime)
-        _swatch = Colors.yellow;
-      else if (_swatch == Colors.yellow)
-        _swatch = Colors.amber;
-      else if (_swatch == Colors.amber)
-        _swatch = Colors.orange;
-      else if (_swatch == Colors.orange)
-        _swatch = Colors.deepOrange;
-      else if (_swatch == Colors.deepOrange)
-        _swatch = Colors.brown;
-      else if (_swatch == Colors.brown)
-        _swatch = Colors.grey;
-      else if (_swatch == Colors.grey)
-        _swatch = Colors.blueGrey;
+      if (_swatch == kSwatches.last) {
+        _swatch = kSwatches.first;
+      } else {
+        // take the next color in _swatches list
+        _swatch = kSwatches
+            .skipWhile((MaterialColor swatch) => swatch != _swatch)
```

Comment:
I'm not expert in complexity evaluation but I don't see any difference between the new version and the old one - every values are compared.

Index added.